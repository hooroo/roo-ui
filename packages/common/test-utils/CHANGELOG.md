# Change Log

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

<a name="0.4.0"></a>
# [0.4.0](https://github.com/hooroo/roo-ui/compare/v0.3.3...v0.4.0) (2018-04-26)


### Features

* add theming ([bee896a](https://github.com/hooroo/roo-ui/commit/bee896a))
