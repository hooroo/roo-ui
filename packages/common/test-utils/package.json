{
  "name": "@roo-ui/test-utils",
  "version": "0.0.0",
  "main": "dist/index.js",
  "publishConfig": {
    "access": "public"
  },
  "scripts": {
    "build": "babel src -d dist --copy-files"
  }
}
