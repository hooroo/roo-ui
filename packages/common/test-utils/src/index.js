export { default as shallowWithTheme } from './shallowWithTheme'; // eslint-disable-line import/prefer-default-export
