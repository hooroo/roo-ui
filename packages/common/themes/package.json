{
  "name": "@roo-ui/themes",
  "version": "0.4.0",
  "main": "dist/index.js",
  "publishConfig": {
    "access": "public"
  },
  "scripts": {
    "build": "babel src -d dist --copy-files"
  }
}
