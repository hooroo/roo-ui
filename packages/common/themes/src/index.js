import qantas from './qantas';

export { qantas }; // eslint-disable-line import/prefer-default-export
