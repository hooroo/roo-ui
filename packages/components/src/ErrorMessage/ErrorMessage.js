import PropTypes from 'prop-types';
import { css } from 'styled-components';
import { themeGet } from 'styled-system';
import { rem } from 'polished';

import Text from '../Text';

const ErrorMessage = Text.extend`
  padding: ${themeGet('space.3')};
  position: relative;
  width: 100%;
  display: block;
  text-align: left;

  &:after {
    content: '';
    height: 0;
    width: 0;
    position: absolute;
    border: solid transparent;
    border-width: ${rem('10px')};
  }

  ${props => props.arrow === 'top' && css`
    &:after {
      bottom: 100%;
      left: ${themeGet('space.3')};
      border-bottom-color: ${themeGet('colors.ui.errorBackground')};
    }
  `}

  ${props => props.arrow === 'left' && css`
    &:after {
      top: 50%;
      right: 100%;
      transform: translateY(-50%);
      border-right-color: ${themeGet('colors.ui.errorBackground')};
    }
  `}

  ${props => props.arrow === 'right' && css`
    &:after {
      top: 50%;
      left: 100%;
      transform: translateY(-50%);
      border-left-color: ${themeGet('colors.ui.errorBackground')};
    }
  `}

  ${props => props.arrow === 'bottom' && css`
    &:after {
      top: 100%;
      left: ${themeGet('space.3')};
      border-top-color: ${themeGet('colors.ui.errorBackground')};
    }
  `}
`;

ErrorMessage.propTypes = {
  arrow: PropTypes.oneOf(['top', 'right', 'bottom', 'left']),
};

ErrorMessage.defaultProps = {
  bg: 'ui.errorBackground',
  arrow: null,
  blacklist: Object.keys(ErrorMessage.propTypes),
};

export default ErrorMessage;

