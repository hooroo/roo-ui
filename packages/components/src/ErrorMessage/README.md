# Error message

<!-- STORY -->

## Installation

```shell
$ yarn add @roo-ui/components
```

## Example

```js
import { ErrorMessage } from '@roo-ui/components';

export default (
  <ErrorMessage>An error occurred</ErrorMessage>
);
```

## Properties

| Name          | Description                | Type   | Default | Required? |
|:--------------|:---------------------------|:-------|:--------|:----------|
| `arrowTop`    | arrow is placed on top     | `bool` | false   | -         |
| `arrowRight`  | arrow is placed on right   | `bool` | false   | -         |
| `arrowBottom` | arrow is placed on bottom  | `bool` | false   | -         |
| `arrowLeft`   | arrow is placed on left    | `bool` | false   | -         |

## Customization

This component can be customized with [styled-system](https://github.com/jxnblk/styled-system) by passing props for [color](https://github.com/jxnblk/styled-system#color-responsive), [font size](https://github.com/jxnblk/styled-system#fontsize-responsive), [space](https://github.com/jxnblk/styled-system#space-responsive), or [typography](https://github.com/jxnblk/styled-system#typography).
