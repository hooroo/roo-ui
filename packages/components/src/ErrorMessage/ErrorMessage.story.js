import React from 'react';
import { storiesOf } from '@storybook/react';
import { withDocs } from 'storybook-readme';
import { select } from '@storybook/addon-knobs/react';

import ErrorMessage from '.';
import README from './README.md';

storiesOf('Components|ErrorMessage', module)
  .addDecorator(withDocs(README))
  .add('default', () => (
    <ErrorMessage
      arrow={select('Arrow', ['top', 'right', 'bottom', 'left'], 'top')}
    >
      An error occurred
    </ErrorMessage>
  ));
