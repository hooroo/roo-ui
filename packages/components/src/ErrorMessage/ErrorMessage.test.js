import React from 'react';
import { qantas as theme } from '@roo-ui/themes';
import { shallowWithTheme } from '@roo-ui/test-utils';

import ErrorMessage from '.';

describe('<ErrorMessage />', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallowWithTheme(<ErrorMessage>An error occurred</ErrorMessage>, theme);
  });

  it('renders correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  describe('arrowTop', () => {
    beforeEach(() => {
      wrapper = shallowWithTheme(<ErrorMessage arrowTop>An error occurred</ErrorMessage>, theme);
    });

    it('renders correctly', () => {
      expect(wrapper).toMatchSnapshot();
    });
  });

  describe('arrowRight', () => {
    beforeEach(() => {
      wrapper = shallowWithTheme(<ErrorMessage arrowRight>An error occurred</ErrorMessage>, theme);
    });

    it('renders correctly', () => {
      expect(wrapper).toMatchSnapshot();
    });
  });

  describe('arrowBottom', () => {
    beforeEach(() => {
      wrapper = shallowWithTheme(<ErrorMessage arrowBottom>An error occurred</ErrorMessage>, theme);
    });

    it('renders correctly', () => {
      expect(wrapper).toMatchSnapshot();
    });
  });

  describe('arrowLeft', () => {
    beforeEach(() => {
      wrapper = shallowWithTheme(<ErrorMessage arrowLeft>An error occurred</ErrorMessage>, theme);
    });

    it('renders correctly', () => {
      expect(wrapper).toMatchSnapshot();
    });
  });
});
