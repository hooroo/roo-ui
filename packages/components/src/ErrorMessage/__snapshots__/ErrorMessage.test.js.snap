// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ErrorMessage /> prop.arrow equals bottom renders correctly 1`] = `
<StyledComponent
  arrow="bottom"
  bg="ui.errorBackground"
  forwardedRef={null}
>
  An error occurred
</StyledComponent>
`;

exports[`<ErrorMessage /> prop.arrow equals left renders correctly 1`] = `
<StyledComponent
  arrow="left"
  bg="ui.errorBackground"
  forwardedRef={null}
>
  An error occurred
</StyledComponent>
`;

exports[`<ErrorMessage /> prop.arrow equals right renders correctly 1`] = `
<StyledComponent
  arrow="right"
  bg="ui.errorBackground"
  forwardedRef={null}
>
  An error occurred
</StyledComponent>
`;

exports[`<ErrorMessage /> prop.arrow equals top renders correctly 1`] = `
<StyledComponent
  arrow="top"
  bg="ui.errorBackground"
  forwardedRef={null}
>
  An error occurred
</StyledComponent>
`;

exports[`<ErrorMessage /> renders correctly 1`] = `
<StyledComponent
  arrow={null}
  bg="ui.errorBackground"
  forwardedRef={null}
>
  An error occurred
</StyledComponent>
`;
